File: android/domain/src/main/java/com/ex/lolcompose/domain/repository/ChampionRepository.kt
package com.ex.lolcompose.domain.repository

import com.ex.lolcompose.domain.common.DataResult
import com.ex.lolcompose.domain.model.Champion
import kotlinx.coroutines.flow.Flow

interface ChampionRepository {
    fun getChampionList(): Flow<DataResult<List<Champion>>>
    fun getChampion(id: String): Flow<DataResult<Champion>>
}

File: react/src/features/champion/domain/repositories/champion-repository.ts
import type { Champion } from '@/features/champion/domain/entities/champion';

export interface ChampionRepository {
  getChampionList(patchVersion: string, locale: string): Promise<readonly Champion[]>;
  getChampion(championId: string, patchVersion: string, locale: string): Promise<Champion>;
}
